### DISCUSSION TITLE
Dynamic Profile: error or missing attribute

### DISCUSSION THREAD

**Discussion by JannekeArntzen** [chris-greening/instascrape#68] [Q&A]
hi Chris, 
Thank you so much for all your work on Instascrape, I really enjoy your blogposts. I have a question about the examples. I read most of the tutorials and looked into the 'Dynamic Profile' class you made. 

In the example of Joe Biden, the code reads:
```
joe = DynamicProfile('https://www.instagram.com/joebiden')
posts = joe.get_all_posts()
```
However, this gives me an error (InvalidArgumentException: invalid argument: 'url' must be a string). 

In other examples profile-scraping, you use 
`profile = DynamicProfile.from_username(username)`
But in the DynamicProfile class, there is no attribute or function defined as 'from_username' (AttributeError: type object 'DynamicProfile' has no attribute 'from_username'). 

Could you please help me out? 

Janneke

**Comment by chris-greening** [comment]
Hey Janneke! 

I'm not able to replicate the InvalidArgumentException for the Joe Biden example and it seems to be working for me, which version of instascrape do you have on your machine? 

I have to go back and reupdate those older blog posts/examples to reflect recent API changes 😅 I've got some massive updates coming down the line that I'll be making to the library, documentation, and blog posts so keep an eye out for those!! 

Thanks so much for reaching out! 

Chris

**Reply by JannekeArntzen** [reply]
Thanks for replying, Chris! I am using instascraper version 1.7.1. I will stay tuned for upcoming changes in instascraper :-)
